src/main/java/games/strategy/engine/lobby/server/db/MutedMacController.java

```diff
@@ -58,14 +59,17 @@ private static void removeMutedMac(final String mac) {
    * database any mac's whose mute has expired.
    */
   public boolean isMacMuted(final String mac) {
-    final long muteTill = getMacUnmuteTime(mac);
-    return muteTill > now().toEpochMilli();
+    final Optional<Instant> muteTill = getMacUnmuteTime(mac);
+    return !muteTill.isPresent() || muteTill.get().isAfter(now());
   }
 
   /**
-   * Returns epoch milli second timestamp of when a mute expires or negative one if there is no mute.
+   * Returns an Optional Instant of the moment when the mute expires.
+   * The optional is empty when the mute never expires.
+   * If the mac isn't muted or the mute already expired the optional instant is
+   * Instant.EPOCH
```

Comment:
Hmm, this is the opposite of what I would have expected.  An empty `Optional` seems like it should represent the "not muted" case.  This confused me when I was trying to figure out why `isMacMuted()` returns `true` when the `Optional` is empty.

Although I can't remember where, I seem to remember discussing using an `Optional<Instant>` return value here before, but my recollection is

* `Optional.empty()` if no mute or mute has expired.
* `Optional.of(value)` for a finite mute.
* `Optional.of(Instant.MAX)` for an infinite mute.

The following changes describe what I'm talking about:

```diff
diff --git a/src/integ_test/java/games/strategy/engine/lobby/server/db/MutedMacControllerIntegrationTest.java b/src/integ_test/java/games/strategy/engine/lobby/server/db/MutedMacControllerIntegrationTest.java
index 3953d9f35..042101c22 100644
--- a/src/integ_test/java/games/strategy/engine/lobby/server/db/MutedMacControllerIntegrationTest.java
+++ b/src/integ_test/java/games/strategy/engine/lobby/server/db/MutedMacControllerIntegrationTest.java
@@ -7,6 +7,7 @@ import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.when;
 
 import java.time.Instant;
+import java.util.Optional;
 
 import org.junit.jupiter.api.Test;
 
@@ -21,41 +22,44 @@ public class MutedMacControllerIntegrationTest {
   public void testMuteMacForever() {
     muteMacForSeconds(Long.MAX_VALUE);
     assertTrue(controller.isMacMuted(hashedMac));
-    assertFalse(controller.getMacUnmuteTime(hashedMac).isPresent());
+    assertEquals(Optional.of(Instant.MAX), controller.getMacUnmuteTime(hashedMac));
   }
 
   @Test
   public void testMuteMac() {
     final Instant muteUntil = muteMacForSeconds(100L);
     assertTrue(controller.isMacMuted(hashedMac));
-    assertEquals(muteUntil, controller.getMacUnmuteTime(hashedMac).get());
+    assertEquals(Optional.of(muteUntil), controller.getMacUnmuteTime(hashedMac));
     when(controller.now()).thenReturn(muteUntil.plusSeconds(1L));
     assertFalse(controller.isMacMuted(hashedMac));
+    assertEquals(Optional.empty(), controller.getMacUnmuteTime(hashedMac));
   }
 
   @Test
   public void testUnmuteMac() {
     final Instant muteUntil = muteMacForSeconds(100L);
     assertTrue(controller.isMacMuted(hashedMac));
-    assertEquals(muteUntil, controller.getMacUnmuteTime(hashedMac).get());
+    assertEquals(Optional.of(muteUntil), controller.getMacUnmuteTime(hashedMac));
     muteMacForSeconds(-10L);
     assertFalse(controller.isMacMuted(hashedMac));
+    assertEquals(Optional.empty(), controller.getMacUnmuteTime(hashedMac));
   }
 
   @Test
   public void testMuteMacInThePast() {
     muteMacForSeconds(-10L);
     assertFalse(controller.isMacMuted(hashedMac));
+    assertEquals(Optional.empty(), controller.getMacUnmuteTime(hashedMac));
   }
 
   @Test
   public void testMuteMacUpdate() {
     muteMacForSeconds(Long.MAX_VALUE);
     assertTrue(controller.isMacMuted(hashedMac));
-    assertFalse(controller.getMacUnmuteTime(hashedMac).isPresent());
+    assertEquals(Optional.of(Instant.MAX), controller.getMacUnmuteTime(hashedMac));
     final Instant muteUntil = muteMacForSeconds(100L);
     assertTrue(controller.isMacMuted(hashedMac));
-    assertEquals(muteUntil, controller.getMacUnmuteTime(hashedMac).get());
+    assertEquals(Optional.of(muteUntil), controller.getMacUnmuteTime(hashedMac));
   }
 
   private Instant muteMacForSeconds(final long length) {
diff --git a/src/main/java/games/strategy/engine/lobby/server/db/MutedMacController.java b/src/main/java/games/strategy/engine/lobby/server/db/MutedMacController.java
index 7fa0e2952..42a3af888 100644
--- a/src/main/java/games/strategy/engine/lobby/server/db/MutedMacController.java
+++ b/src/main/java/games/strategy/engine/lobby/server/db/MutedMacController.java
@@ -59,15 +59,12 @@ public class MutedMacController extends TimedController {
    * database any mac's whose mute has expired.
    */
   public boolean isMacMuted(final String mac) {
-    final Optional<Instant> muteTill = getMacUnmuteTime(mac);
-    return !muteTill.isPresent() || muteTill.get().isAfter(now());
+    return getMacUnmuteTime(mac).map(instant -> instant.isAfter(now())).orElse(false);
   }
 
   /**
    * Returns an Optional Instant of the moment when the mute expires.
-   * The optional is empty when the mute never expires.
-   * If the mac isn't muted or the mute already expired the optional instant is
-   * Instant.EPOCH
+   * The optional is empty when the mac is not muted or the mute has already expired.
    */
   public Optional<Instant> getMacUnmuteTime(final String mac) {
     final String sql = "select mac, mute_till from muted_macs where mac=?";
@@ -80,7 +77,7 @@ public class MutedMacController extends TimedController {
         if (found) {
           final Timestamp muteTill = rs.getTimestamp(2);
           if (muteTill == null) {
-            return Optional.empty();
+            return Optional.of(Instant.MAX);
           }
           final Instant expiration = muteTill.toInstant();
           if (expiration.isBefore(now())) {
@@ -88,11 +85,11 @@ public class MutedMacController extends TimedController {
             // If the mute has expired, allow the mac
             removeMutedMac(mac);
             // Signal as not-muted
-            return Optional.of(Instant.EPOCH);
+            return Optional.empty();
           }
           return Optional.of(expiration);
         }
-        return Optional.of(Instant.EPOCH);
+        return Optional.empty();
       }
     } catch (final SQLException sqle) {
       throw new IllegalStateException("Error for testing muted mac existence:" + mac, sqle);
```

Is there something I'm missing about reversing the meaning of an empty `Optional`?